openlayers/openlayers.d.ts

```diff
@@ -4085,76 +4085,117 @@ declare namespace ol {
 
     namespace source {
 
-        class BingMaps {
+        class BingMaps extends TileImage {
         }
 
-        class Cluster {
+        class CartoDB extends XYZ {
         }
 
-        class Image {
+        class Cluster extends Vector {
+        }
+
+        class Image extends Source {
         }
 
-        class ImageCanvas {
+        class ImageArcGISRest extends Image {
+        }
+
+        class ImageCanvas extends Image {
         }
 
         class ImageEvent {
         }
 
-        class ImageMapGuide {
+        class ImageMapGuide extends Image {
         }
 
-        class ImageStatic {
+        class ImageStatic extends Image {
         }
 
-        class ImageVector {
+        class ImageVector extends ImageCanvas {
         }
 
-        class ImageWMS {
+        class ImageWMS extends Image {
             constructor(options: olx.ImageWMSOptions);
         }
 
-        class MapQuest {
+        class MapQuest extends XYZ {
             constructor(options: any);
         }
 
-        class OSM {
+        class OSM extends XYZ {
         }
 
-        class Source {
+        class Source extends Object {
+            /**
+             * Get the projection of the source.
+             * @return Projection.
+             */
+            getProjection(): ol.proj.Projection;
+
+            /**
+             * Refreshes the source and finally dispatches a 'change' event.
+             */
+            refresh(): void;
         }
 
-        class Stamen {
+        class Stamen extends XYZ {
         }
 
-        class Tile {
+        class Tile extends Source {
         }
 
-        class TileArcGISRest {
+        class TileArcGISRest extends TileImage {
         }
 
-        class TileDebug {
+        class TileDebug extends Tile {
         }
 
         class TileEvent {
         }
 
-        class TileImage {
+        class TileImage extends UrlTile {
         }
 
-        class TileJSON {
+        class TileJSON extends TileImage {
         }
 
-        class TileUTFGrid {
+        class TileUTFGrid extends Tile {
         }
 
-        class TileVector {
+        class TileWMS extends TileImage {
+            constructor(options: olx.TileWMSOptions);
+
+            /**
+             * Update the user-provided (WMS request) parameters.
+             */
+            updateParams(params: {}): void;
+
+            /**
+             * Get the user-provided (WMS request) params, i.e. those passed to the constructor through the "params" option, and possibly updated using the updateParams method.
+             */
+            getParams(): {};
```

Comment:
this looks like an opaque object passed by the user, i would expect this to be typed as `any`.